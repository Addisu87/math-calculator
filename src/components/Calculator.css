body {
  font-family: helvetica, arial, sans-serif;
}

button {
  font-family: inherit;
  background-color: grey;
  color: #000;
  font-size: 1.5em;
  padding: 25px 33px;
}
