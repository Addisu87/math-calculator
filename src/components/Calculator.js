import React, { useState } from 'react';
import styled from 'styled-components';
import RowDisplay from './RowDisplay';
import calculate from '../logic/calculate';
import Button from './Button';

function Calculator() {
  const [state, setState] = useState({
    next: null,
    total: null,
    operation: null,
  });
  const handleClick = (e) => {
    setState((state) => calculate(
      {
        next: state.next,
        total: state.total,
        operation: state.operation,
      },
      e.target.value,
    ));
  };

  const { next, total, operation } = state;
  return (
    <Wrapper>
      <h5 className="doMath">Let&apos;s do some math!</h5>

      <CalcContainer>
        <RowDisplay next={next} total={total} operation={operation} />

        <div className="rowBtn normal">
          <Button type="button" label="AC" onClick={handleClick} />
          <Button type="button" label="+/-" onClick={handleClick} />
          <Button type="button" label="%" onClick={handleClick} />
          <Button type="operator" label="รท" onClick={handleClick} />
        </div>

        <div className="rowBtn normal">
          <Button type="button" label="7" onClick={handleClick} />
          <Button type="button" label="8" onClick={handleClick} />
          <Button type="button" label="9" onClick={handleClick} />
          <Button type="operator" label="x" onClick={handleClick} />
        </div>

        <div className="rowBtn normal">
          <Button type="button" label="4" onClick={handleClick} />
          <Button type="button" label="5" onClick={handleClick} />
          <Button type="button" label="6" onClick={handleClick} />
          <Button type="operator" label="-" onClick={handleClick} />
        </div>

        <div className="rowBtn normal">
          <Button type="button" label="1" onClick={handleClick} />
          <Button type="button" label="2" onClick={handleClick} />
          <Button type="button" label="3" onClick={handleClick} />
          <Button type="operator" label="+" onClick={handleClick} />
        </div>

        <div className="rowBtn double">
          <Button type="button" label="0" onClick={handleClick} />
          <Button type="button" label="." onClick={handleClick} />
          <Button type="operator" label="=" onClick={handleClick} />
        </div>
      </CalcContainer>
    </Wrapper>
  );
}

export default Calculator;

const Wrapper = styled.div`
  display: flex;
  & .doMath {
    flex: 1;
  }
`;

const CalcContainer = styled.div`
  background-color: var(--bg-color);
  height: 70vh;
  padding: 10px;
  width: 70%;
`;
