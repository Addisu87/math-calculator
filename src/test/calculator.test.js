import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('should render without problems', () => {
  it('should render the calculator', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
