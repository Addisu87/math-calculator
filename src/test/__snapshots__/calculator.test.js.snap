// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without problems should render the calculator 1`] = `
<div
  className="sc-bczRLJ bptONv"
>
  <h5
    className="doMath"
  >
    Let's do some math!
  </h5>
  <div
    className="sc-gsnTZi ejTGak"
  >
    <div
      className="screen-row"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="rowBtn normal"
    >
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        className="btn-operator"
        onClick={[Function]}
        type="button"
        value="รท"
      />
    </div>
    <div
      className="rowBtn normal"
    >
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        className="btn-operator"
        onClick={[Function]}
        type="button"
        value="x"
      />
    </div>
    <div
      className="rowBtn normal"
    >
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        className="btn-operator"
        onClick={[Function]}
        type="button"
        value="-"
      />
    </div>
    <div
      className="rowBtn normal"
    >
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        className="btn-operator"
        onClick={[Function]}
        type="button"
        value="+"
      />
    </div>
    <div
      className="rowBtn double"
    >
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        className="btn-button"
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        className="btn-operator"
        onClick={[Function]}
        type="button"
        value="="
      />
    </div>
  </div>
</div>
`;
